packages/tooltip/src/components/tooltip/Tooltip.tsx

```diff
@@ -1,14 +1,84 @@
-import { Text } from "@stenajs-webui/core";
+import { Indent, Row, Text } from "@stenajs-webui/core";
+import {
+  Icon,
+  stenaExclamationCircle,
+  stenaExclamationTriangle,
+  stenaInfoCircle,
+} from "@stenajs-webui/elements";
+import TippyComponent, {
+  TippyProps as TippyComponentProps,
+} from "@tippyjs/react";
+import cx from "classnames";
 import * as React from "react";
-import { Popover, PopoverProps, PopoverVariant } from "../popover/Popover";
+import { TippyCallbackRef } from "../../hooks/UseTippyInstance";
+import styles from "./Tooltip.module.css";
 
-export interface TooltipProps extends Partial<Omit<PopoverProps, "content">> {
-  variant?: PopoverVariant;
+type TooltipVariant = "info" | "warning" | "error";
+
+interface TippyProps
+  extends Partial<Omit<TippyComponentProps, "theme" | "render">> {
+  tippyRef?: TippyCallbackRef<HTMLDivElement>;
+}
+
+export interface TooltipProps extends TippyProps {
+  variant?: TooltipVariant;
   label: string;
 }
 
-export const Tooltip: React.FC<TooltipProps> = ({ label, ...popoverProps }) => {
+const variantIcons = {
+  info: stenaInfoCircle,
+  warning: stenaExclamationCircle,
+  error: stenaExclamationTriangle,
+};
+
+export const Tooltip: React.FC<TooltipProps> = ({
+  visible,
+  trigger = "mouseenter",
+  children,
+  tippyRef,
+  delay = 0,
+  variant,
+  content,
+  maxWidth = "125px",
+  arrow = true,
+  label,
+  ...tippyProps
+}) => {
+  const textComponent = (
```

Comment:
Another way of doing is to make another component outside of Tooltip (e.g. `const InnerText = ...`) and then use it like `<InnerText label={label}>`. That's how I would do something like this.

But also, maybe this isn't the way to go at all. More on that later.